src/avt/Filters/avtThresholdFilter.C

```diff
@@ -429,9 +527,87 @@ avtThresholdFilter::ProcessOneChunk(avtDataRepresentation *in_dr, bool fromChunk
     if (curOutDataSet != NULL)
         curOutDataSet->Delete();
 
+    visitTimer->StopTimer(timerHandle, "avtThresholdFilter::ProcessOneChunk_VTK");
+    
+    return out_dr;
+}
+
+
+// ****************************************************************************
+//  Method: avtThresholdFilter::ProcessOneChunk_VTKM
+//
+//  Purpose:
+//      Perform threshold using VTKm
+//
+//  Programmer: James Kress
+//  Creation:   November 18, 2020
+//
+//  Modifications:
+//
+// ****************************************************************************
+
+avtDataRepresentation *
+avtThresholdFilter::ProcessOneChunk_VTKM(avtDataRepresentation *in_dr)
+{
+#ifndef HAVE_LIBVTKH
+    return NULL;
+#else
+    int timerHandle = visitTimer->StartTimer();
+    
+    vtkh::DataSet *in_ds = in_dr->GetDataVTKm();
+    if (!in_ds || in_ds->GetNumberOfDomains() != 1)
+        return NULL;
+
+    const stringVector curVariables    = atts.GetListedVarNames();
+    const intVector    curZonePortions = atts.GetZonePortions();
+    const doubleVector curLowerBounds  = atts.GetLowerBounds();
+    const doubleVector curUpperBounds  = atts.GetUpperBounds();
+    const stringVector curBoundsRange = atts.GetBoundsRange();
+    
+    const char *curVarName;
+    char errMsg[1024];
+    
+    vtkh::DataSet *out_ds = in_ds;
+    for (size_t curVarNum = 0; curVarNum < curVariables.size(); curVarNum++)
+    {
+        vtkh::Threshold thresher;
+    
+        std::map<std::string,int>::iterator iterFind;
+        bool bypassThreshold = false;
+        iterFind = selIDs.find(curVariables[curVarNum]);
+        if (iterFind != selIDs.end())
+        {
+            int selID = iterFind->second;
+            if (GetInput()->GetInfo().GetAttributes().GetSelectionApplied(selID))
+            {
+                debug1 << "Bypassing Threshold operator because the database "
+                       << "plugin claims to have applied selection on "
+                       << curVariables[curVarNum] << endl;
+                bypassThreshold = true;
+            }
+        }
+
+        if (bypassThreshold == false)
+        {
+            curVarName = curVariables[curVarNum].c_str();
+            thresher.SetInput(in_ds);
+            thresher.SetField(curVariables[curVarNum]);
```

Comment:
The Threshold operator support thresholding by PartOfZone or EntireZone. This isn't being set, so I assume it is doing one or the other. The best option would be to use the VTK path for the unsupported case.

Reply:
Added to the VTKmAble call. Currently vtkm only supports allInRange